File: 830.较大分组的位置.ts
/*
 * @lc app=leetcode.cn id=830 lang=typescript
 *
 * [830] 较大分组的位置
 */

// @lc code=start
function largeGroupPositions(s: string): number[][] {
    let ret: number[][] = [];
    let start = 0;
    while (start < s.length) {
        let end = start + 1;
        for (;end < s.length && s[start] === s[end]; end++) {}
        if (end - start >= 3) ret.push([start, end-1]);
        start = end;
    }
    return ret;
};
// @lc code=end

File: leetcode/830.较大分组的位置.go
/*
 * @lc app=leetcode.cn id=830 lang=golang
 *
 * [830] 较大分组的位置
 */

// @lc code=start
func largeGroupPositions(s string) (ret [][]int) {
	start := 0
	for start < len(s) {
		end := start + 1
		for ; end < len(s) && s[start] == s[end]; end++ {}
		if end - start >= 3 {
			ret = append(ret, []int{start, end-1})
		}
		start = end
	}
	return
}
// @lc code=end

